DOC-2030: Add documentation for new Spell Checker Pro features (#2759)
tinymce/tinymce-docs: modules/ROOT/pages/introduction-to-tiny-spellchecker.adoc

});
----

=== SpellcheckerUpdated Event

> NOTE: Requires 6.5.
> 
> This event is fired when the editor is spellchecked either by opening the dialog or by executing the `+mceSpellcheckUpdate+` command.

Reviewer: ```suggestion
include::partial$misc/admon-requires-6.5v.adoc[]

This event is fired when the editor is spellchecked: either by opening the Spell Checker Pro dialog, or by executing the `+mceSpellcheckUpdate+` command.
```
1. In anticipation of [PR2761](https://github.com/tinymce/tinymce-docs/pull/2761) being approved and merged.
2. copy-edits and a qualifier for specificity’s sake.

Author: I have some more changes ready here so I'll include this once that PR is merged

> NOTE: This event is not fired in As-You-Type mode.

Reviewer: ```suggestion
NOTE: This event is not fired when SpellChecker Pro is operating in As-You-Type mode.
```
Another edit for specificity.

==== Example: the SpellcheckerLanguageChanged event

[source,js]
----
tinymce.init({
  selector: 'textarea',
  plugins: 'tinymcespellchecker',
  toolbar: 'spellchecker',
  init_instance_callback: (editor) => {
    editor.on('SpellcheckerUpdated', (e) => {
      const spelling = e.spelling;
      for (const [ language, words ] of Object.entries(spelling)) {
        for (const [ word, suggestions ] of Object.entries(words)) {
          console.log(`${language} suggestions for ${word}:`, suggestions);
        }
      }
    });
  }
});
----

==== The `spelling` object

> The `+spelling+` object, provided by the `+SpellcheckerUpdated+` event, contains the result of the spellcheck by the spelling service. The keys of the outermost object represent the language code for each language. Under each language code, there is an object with keys representing misspelled words and values that are arrays of possible correct spellings.

Reviewer: ```suggestion
The `+spelling+` object, which is provided by the `+SpellcheckerUpdated+` event, contains the result of the spelling service spellcheck.

The keys of the outermost object are the language code for each checked language.

Under each returned language code is an object with

* keys representing misspelled words; and
* arrays of values that are possible correct spellings.
```

Copy edits and suggested presentation edits.
